src/Microsoft.OpenApi.OData.Reader/Operation/DollarCountGetOperationHandler.cs

```diff
@@ -75,19 +88,85 @@ protected override void SetResponses(OpenApiOperation operation)
             base.SetResponses(operation);
         }
 
-        protected override void AppendCustomParameters(OpenApiOperation operation)
+        /// <inheritdoc/>
+        protected override void SetParameters(OpenApiOperation operation)
         {
-            IEdmVocabularyAnnotatable annotatable = null;            
-            if (LastSecondSegment is ODataNavigationSourceSegment sourceSegment)
+            base.SetParameters(operation);
```

Comment:
this should also support custom defined request headers like Consistency Level defined on the parent.

Reply:
We already add custom header here: https://github.com/microsoft/OpenAPI.NET.OData/blob/e7bad1197d1469d2cbffaebdefae470fb7cc7043/src/Microsoft.OpenApi.OData.Reader/Operation/DollarCountGetOperationHandler.cs#L103-L106